### PR TITLE
Provide a specific error to raise_error

### PR DESCRIPTION
**Pull request by allenwq**
This removes the warnings from Rspec.

### CODE REVIEW

File: spec/models/course/level_spec.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
   context 'before level_number is set' do
     it 'raises' do
-      expect { Course::Level.new.level_number }.to raise_error
+      expect { Course::Level.new.level_number }.to raise_error(RuntimeError)
```
Comment (lowjoel) [inline spec/models/course/level_spec.rb]: @kxmbrian RuntimeError is bad. Can this be more specific?

Comment (lowjoel) [inline spec/models/course/level_spec.rb]: @allenwq @kxmbrian These errors in Ruby seem suitable:
 - ArgumentError
 - IndexError
 - RangeError

Comment (allenwq) [inline spec/models/course/level_spec.rb]: Actually this is not relevant to Argument nor Index, shall we keep RuntimeError or define a new error ?

Comment (lowjoel) [inline spec/models/course/level_spec.rb]: I’m open to defining a new error. Need a name though.

Comment (allenwq) [inline spec/models/course/level_spec.rb]: How about `LevelNotComputedError`

Comment (lowjoel) [inline spec/models/course/level_spec.rb]: Too specific. This sounds like an object being in an invalid state.

Comment (allenwq) [inline spec/models/course/level_spec.rb]: If need to be general, `StateInvalidError`?

Comment (lowjoel) [inline spec/models/course/level_spec.rb]: C# calls it InvalidOperationException, Java calls it IllegalStateException. I would call it something along those lines.

**Pull request merged**